DataFormats/TrackReco/interface/HitPattern.h

```diff
@@ -262,9 +262,9 @@ class HitPattern
     void printHitPattern(HitCategory category, int position, std::ostream &stream) const;
     void print(HitCategory category, std::ostream &stream = std::cout) const;
 
-    bool hasValidHitInFirstPixelBarrel() const; // has valid hit in PXB layer 1
-    bool hasValidHitInFirstPixelEndcap() const; // has valid hit in PXF layer 1
-
+    // has valid hit in PXB/PXF layer x 
+    bool hasValidHitInPixelLayer(bool inBarrel, uint16_t layer) const;
+    
```

Comment:
Despite of the +1 by @VinInn let me express a concern on the interface. Using a `bool` to denote barrel/endcap is not very clear when looking the caller code alone ("Was true for barrel or for endcap?"). May I suggest to use e.g. an `enum` instead?